CHANGELOG.rst

```diff
@@ -6,12 +6,22 @@
 
 - The ancestors tree sequence now contains the real alleles and not
   0/1 values as before.
+
 - Times for undated sites now use frequencies (0..1), not as counts (1..num_samples),
   and are now stored as -inf, then calculated on the fly in the variants() iterator.
+
 - The SampleData file no longer accepts the ``inference`` argument to add_site.
   This functionality has been replaced by the ``exclude_positions`` argument
   to the ``infer`` and ``generate_ancestors`` functions.
 
+**New features**:
```

Comment:
That's a breaking change, surely.

Reply:
I don't think it breaks anything, to be honest. We can still read-in sample data files that have `{}` in the metadata file. The reason that I would like to do this is that I want to round-trip data (specifically metadata) from a tree sequence through `.from_tree_sequence()`, via inference, and back to another tree sequence, and this eases that transition, I think.

Reply:
It can break downstream code that depends on the ts having a dictionary in the metadata, not None.

Reply:
Fair enough - I can add it to breaking changes. But the JSON codec for tree sets allows this (and it's the default), so I think any such code may be problematic anyway (and will stop working when we switch to defaulting to the JSON codec for inferred tree sequences, which I assume we want to do?

Reply:
I just think there's no point in doing this and making the change before we've added the schemas - we're adding extra breakage that we'll have to keep track of and document.

So, add the schemas first. We'll have to do this within a week anyway, so there's no point in putting it off.

Reply:
Well, if we are doing it this week, then sure!